annex/chapter1/adv_examples/src/example8.cpp

```diff
@@ -0,0 +1,47 @@
+/****************************************************************************
+*    1 event loop
+*    Publish and subscriber for other types of data: my own struct Position(x, y)
+*    Distance to origin computation
+*****************************************************************************/
+
+#include <iostream>
+#include <klepsydra/high_performance/event_loop_middleware_provider.h>
+#include <thread>
+#include <chrono>
+#include "distance_calculation.h"
+
+int main() {
+    kpsr::high_performance::EventLoopMiddlewareProvider<16> eventloop(nullptr);
+
+    kpsr::Publisher<Position> * positionPublisher = eventloop.getPublisher<Position>("whereabout", 0, nullptr, nullptr);
+    
+
+    eventloop.start();
+
+    {
+        DistanceCalculation distanceCalculation(eventloop.getSubscriber<Position>("whereabout"), eventloop.getPublisher<float>("distanceOrigin", 0, nullptr, nullptr));
+    
+        eventloop.getSubscriber<float>("distanceOrigin")->registerListener("distanceOrigin", [](const float & message) {
+            std::cout << "Distance to origin : " << message << std::endl;
+            std::thread::id this_id = std::this_thread::get_id();
+            std::cout << "the distanceListener id is : " << this_id << std::endl;
+
+            });
+
+        std::thread t([&positionPublisher]() {
```

Comment:
As @pablogh-2000 mentioned in his review, change the thread name to something more expressive. Make the change in this and in other instances also (i.e t1, t2, t3 etc in this file and in example9)

Reply:
Done